### ISSUE TITLE
Support depends_on type service_completed_successfully

### ISSUE THREAD

**Issue by micheljung** [containers/podman-compose#575]
**Is your feature request related to a problem? Please describe.**

The problem is that podman-compose is incompatible with current docker-compose.yml because it does not support `service_completed_successfully`.

According to the [compose spec](https://github.com/compose-spec/compose-spec/blob/master/spec.md#depends_on)

- `condition`: condition under which dependency is considered satisfied
  - `service_started`: is an equivalent of the short syntax described above
  - `service_healthy`: specifies that a dependency is expected to be "healthy"
    (as indicated by [healthcheck](#healthcheck)) before starting a dependent
    service.
  - `service_completed_successfully`: specifies that a dependency is expected to run
    to successful completion before starting a dependent service.

**Describe the solution you'd like**
podman-compose implements `service_completed_successfully` as per the spec.

**Describe alternatives you've considered**
None

**Additional context**
I didn't check if the other conditions are supported yet, but they should also be.

**Comment by mbergen** [comment]
It seems like `service_healthy` also does not work.
`service_started` is the equivalent to no condition (default) and therefore works.
Would be great to get these working!

**Comment by mdellweg** [comment]
Actually, i think podman waits for the healthiness anyway, so anything one specifies behaves like `service_healthy`.

**Comment by mbergen** [comment]
> Actually, i think podman waits for the healthiness anyway, so anything one specifies behaves like `service_healthy`.

Last time i checked it podman was not executing any healthchecks, it was only waiting for services to be started. That's a huge difference.

**Comment by Greenall** [comment]
I'm on the following versions and can confirm that service_healthy condition seems to be ignored and treated as service_started. 
podman-compose version 1.0.6
podman version 4.4.1

**Comment by kontsaki** [comment]
any status update on this feature?

any way to workaround this?

**Comment by w568w** [comment]
@kontsaki Have a look at #453, which has a preliminary implementation of `depends_on`.

**TL;DR (maybe inaccurate!):** `podman-compose` is calling `podman` commands directly to do all stuffs (e.g. executing `podman start foo --bar`), which do not provide a way of "depending on" another container, because it needs a daemon to do that (as Docker does). There are some workarounds:

- `podman-compose` waits for all dependencies to start (dirty, but has been done in #453)
- rely on `podman` socket and use its API to start instead of calling it directly (see discussion in <https://github.com/containers/podman-compose/discussions/456>; community has no motivation to do this)
- wait until Podman officially supports flags like `--require bar` (almost no way)

So, it is sad to say that there is still no good way to support it.

**Comment by kontsaki** [comment]
I ended up using docker-compose provided by https://github.com/docker/compose and it works as expected with podman as the backend. 

I have also installed the compat package podman-docker on Fedora, not sure if this makes podman run as daemon but currently everything works with rootless podman and all docker-compose features.

**Comment by candleindark** [comment]
@mbergen, @mdellweg, @Greenall I just filed a [dedicated issue](https://github.com/containers/podman-compose/issues/866) for the problem with the `service_healthy` condition.

**Comment by procopym** [comment]
Is there any progress on implementation of depends_on type service_completed_successfully?

**Comment by prabhjotsbhatia-ca** [comment]
Narrowed down onto this issue as well, after wasting several (embarassed!) hours troubleshooting.

**Comment by jaredmontoya** [comment]
When running trying to run the stack made for docker compose I have to manually restart some containers multiple times for podman compose to recognize their current state and continue the execution.

**Comment by mokibit** [comment]
This issue appears to have been fixed by https://github.com/containers/podman-compose/pull/1481.

@p12tic this issue can be closed.

**Comment by p12tic** [comment]
Fixed in https://github.com/containers/podman-compose/pull/1481.
